src/app/problems/problems.module.ts

```diff
@@ -68,6 +70,8 @@ import {DragulaModule} from "ng2-dragula";
         DragulaModule.forRoot(),
     ],
     providers: [
+        DifficultyService,
```

Comment:
Why didn't you add all the providers here?

Reply:
The stuff I searched up showed that you didn't need to add the services that are only used within the module to the providers.